cocotb/handle.py

```diff
@@ -532,10 +521,21 @@ def _range_iter(self, left, right):
     def value(self):
         # need to iterate over the sub-object
         result = []
-        for x in range(len(self)):
-            result.append(self[x].value)
+        for sub_handle in self:
+            result.append(sub_handle.value)
         return result
```

Comment:
Perhaps while you're here, just:
```suggestion
return [sub_handle.value for sub_handle in self]
```